databroker/core.py

```diff
@@ -193,6 +196,19 @@ def __getstate__(self):
                               for k, arg in self._captured_init_kwargs.items()})
         return OrderedDict(cls=self.classname, args=args, kwargs=kwargs)
 
+    def get(self, **user_parameters):
+        if self._data_source is None:
```

Comment:
The second time `get` is called (and thereafter) the `user_parameters` will be ignored and we will get the result from whatever `user_parameters` were given the first time `get` was called.

The `Entry` class just above attempts to do this caching in a way that accounts for the `user_parameters` a.k.a `kwargs`. Would it work to make `StreamEntry` inherit from `Entry` and reuse its mechanism?

Reply:
I dont think the LRU cache will work for this.
I think you want to permanently cache the streams or else you can have the same issue again.

Reply:
Updated